pandas/tests/indexing/test_iloc.py

```diff
@@ -1186,6 +1186,14 @@ def test_iloc_setitem_series_duplicate_columns(self):
         df.iloc[:, 0] = df.iloc[:, 0].astype(np.float64)
         assert df.dtypes.iloc[2] == np.int64
 
+    def test_iloc_setitem_dtypes_duplicate_columns(self):
+        df = DataFrame([["0", "str", "1.2"]], columns=["a", "b", "b"])
+        df.iloc[:, 0] = df.iloc[:, 0].astype("int64")
+        assert df.dtypes.iloc[0] == np.int64
```

Comment:
Please parametrize and check df with tm.assert_frame_equal and also add gh reference

Reply:
Added tm.assert_frame_equal check and gh reference.